e2e/app/google-login.ts

```diff
@@ -23,41 +23,51 @@ export default class GoogleLoginPage extends BasePage {
    * Login email input field.
    */
   async email(): Promise<ElementHandle> {
-    return await this.page.waitForXPath(selectors.emailInput, {visible: true});
+    return this.page.waitForXPath(selectors.emailInput, {visible: true});
   }
 
   /**
    * Login password input field.
    */
   async password(): Promise<ElementHandle> {
-    return await this.page.waitForXPath(selectors.passwordInput, {visible: true});
+    return this.page.waitForXPath(selectors.passwordInput, {visible: true});
   }
 
   /**
    * Google login button.
    */
   async loginButton(): Promise<ElementHandle> {
-    return await this.page.waitForXPath(selectors.loginButton, {visible: true});
+    return this.page.waitForXPath(selectors.loginButton, {visible: true});
   }
 
   /**
    * Enter login email and click Next button.
    * @param email
    */
   async enterEmail(userEmail: string) : Promise<void> {
-    let emailInput: ElementHandle;
-    try {
-      emailInput = await this.email()
-    } catch(e) {
-      const randomLink = await this.page.$x('//*[@role="link"]//*[text()="Use another account"]');
-      if (randomLink.length > 0) {
-        await randomLink[0].click();
-      }
-      emailInput = await this.email()
+    // Handle Google "Use another account" dialog if it exists
+    const useAnotherAccountXpath = '//*[@role="link"]//*[text()="Use another account"]';
+    const elemt1 = await Promise.race([
+      this.page.waitForXPath(selectors.emailInput, {visible: true, timeout: 60000}),
+      this.page.waitForXPath(useAnotherAccountXpath, {visible: true, timeout: 60000}),
+    ]);
+
+    // compare to the Use another account link
+    const [link] = await this.page.$x(useAnotherAccountXpath);
+    const isLink = await page.evaluate((e1, e2) => e1 === e2, elemt1, link);
+    if (isLink) {
+      // click " Use another Account " link
+      await Promise.all([
+        this.page.waitForNavigation(),
+        link.click(),
+      ]);
     }
```

Comment:
Unrelated to jest-puppeteer. This change owns thanks to the unpredictable Google authentication. Test won't fail on Google login page.